src/main/java/com/thebois/views/game/ViewableResource.java

```diff
@@ -0,0 +1,69 @@
+package com.thebois.views.game;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import com.badlogic.gdx.graphics.Color;
+import com.badlogic.gdx.graphics.Texture;
+
+import com.thebois.Pawntastic;
+import com.thebois.models.world.resources.ResourceType;
+import com.thebois.views.TextureUtils;
+
+/**
+ * Viewable Resource that correspond to TerrainType but has view related information.
+ */
+public enum ViewableResource {
+    /**
+     * Viewable Water tile.
+     */
+    WATER(
+        ResourceType.WATER,
+        TextureUtils.createSquareTexture((int) Pawntastic.TILE_SIZE, Color.valueOf("#1E90FF"))),
```

Comment:
The textures themselves probably shouldn't be created in the enum, since they're unmanaged resources. It's better to store the color in the enum. The tile size is public static so no need to store it.

Reply:
Resource view should be the only view using the enum, so that shouldn't happen. That is why we have separated our views.